docs/_layouts/default.html

```diff
@@ -90,6 +90,32 @@
   <div id="fb-root"></div>
 
   <script>
+    (function() {
+      var selector = '.inner-content h2, .inner-content h3, .inner-content h4';
+      var elements = document.querySelectorAll(selector);
+      for (var i = 0; i < elements.length; i++) {
+        var elementID;
+
+        if (elements[i].hasAttribute('id')) {
+          elementID = elements[i].getAttribute('id');
+        } else {
+          var textMethod = document.body.textContent ? "textContent" : "innerText";
+          var roughText = elements[i][textMethod];
+
+          // Regex rule for making the title URL-friendly.
+          tidyText = roughText.toLowerCase()
+                              .replace(/\s+/g, '-')
+                              .replace(/[^A-Za-z0-9\-_.]/g, '');
+
+          elements[i].setAttribute('id', tidyText);
```

Comment:
Do the anchors actually work, since you're setting the ID after it's rendered? Does the browser correctly scroll down to the anchor when you open a fresh tab and go to a URL with an anchor?